### DISCUSSION TITLE
Test Executors nunitTestCases does not respect filter's

### DISCUSSION THREAD

**Discussion by mh884** [nunit/nunit3-vs-adapter#933] [Q&A] [answered]
Why Does this line show all the loaded test from assembly?
Is there a way we can show number of test case that respect test filter's before running the test's? 

https://github.com/nunit/nunit3-vs-adapter/blob/fbf83acd7e4ffb1f5fbb6bae71070a98ed9429bd/src/NUnitTestAdapter/NUnit3TestExecutor.cs#L257

**Comment by OsirisTerje** [comment, marked as answer]
Please elaborate

The loadedTestCases.Count is that number.

**Reply by mh884** [reply]
I assume loadedTestCases.Count should be that number

Ya sure i will give an example:
let's say we have total number of test's 10 test cases and we have set of filter's that will target one test case 

Expected to see in the log:
NUnit3TestExecutor converted 1 of 10 NUnit test cases
Actually:
NUnit3TestExecutor converted 10 of 10 NUnit test cases

Command:
 vstest.console.exe XXX\TestProj.dll /TestCaseFilter:"TestCategory=bug"
Result:
 Nunit run one test case

![image](https://user-images.githubusercontent.com/4064964/146977047-1ba2ae43-8b21-49ef-b7d3-a7e81522871b.png)

In case there is no matching test case we seen an 
"Skipping assembly - no matching test cases found"
![image](https://user-images.githubusercontent.com/4064964/146978244-b57fbf90-9d3d-4cec-8bfd-676c0d78eaad.png)

**Reply by OsirisTerje** [reply]
Can you try this with the latest 4.0.0-beta.4 ?

**Reply by mh884** [reply]
The last version solved the issue thanks

**Reply by OsirisTerje** [reply]
Awesome!
